crates/steel-core/src/values/json_vals.rs

```diff
@@ -1,43 +1,49 @@
+use crate::rvals::SteelString;
 use crate::values::lists::List;
 use crate::{
     gc::Gc,
     rerrs::SteelErr,
     rvals::{FromSteelVal, IntoSteelVal, Result, SteelVal},
-    throw,
 };
 use im_rc::HashMap;
 use serde_json::{Map, Number, Value};
 use std::convert::{TryFrom, TryInto};
-
-pub fn string_to_jsexpr() -> SteelVal {
-    SteelVal::FuncV(|args: &[SteelVal]| -> Result<SteelVal> {
-        if args.len() != 1 {
-            stop!(ArityMismatch => "string->jsexpr takes 1 argument");
-        } else {
-            let arg =
-                &args[0].string_or_else(throw!(TypeMismatch => "string->jsexpr takes a string"))?;
-            // let unescaped = unescape(arg);
-            // let res: std::result::Result<Value, _> = serde_json::from_str(unescaped.as_str());
-            let res: std::result::Result<Value, _> = serde_json::from_str(arg);
-            match res {
-                Ok(res) => res.try_into(),
-                Err(e) => stop!(Generic => format!("string->jsexpr failed: {e}")),
-            }
-        }
-    })
+use steel_derive::function;
+
+/// Deserializes a JSON string into a Steel value.
+///
+/// (string->jsexpr json) -> any/c?
+///
+/// * json : string?
+///
+/// # Examples
+/// ```scheme
+/// (string->jsexpr "{\"foo\": [3]}") ;; => '#hash((foo . (3)))
+/// ```
+#[function(name = "string->jsexpr")]
+pub fn string_to_jsexpr(value: &SteelString) -> Result<SteelVal> {
```

Comment:
I've changed this to use the derive macro, not sure if it has any undesired consequences.

Reply:
This should be fine, the derive macro is in general the preferred way of implementing it anyway